api/public/public_api/endpoints/google_stt.py

```diff
@@ -107,25 +115,30 @@ def _build_response(self, stt_response):
         the case when send a ambiguous voice file and the correct utterance is
         not the utterance with highest confidence and the API.
         """
-        limit = int(self.request.args['limit'])
+        limit = int(self.request.args["limit"])
         if isinstance(stt_response, dict):
             alternative = stt_response.get("alternative")
-            if 'confidence' in alternative:
+            if "confidence" in alternative:
                 # Sorting by confidence:
                 alternative = sorted(
-                    alternative,
-                    key=lambda alt: alt['confidence'],
-                    reverse=True
+                    alternative, key=lambda alt: alt["confidence"], reverse=True
                 )
-                alternative = [alt['transcript'] for alt in alternative]
+                alternative = [alt["transcript"] for alt in alternative]
                 # client is interested in test the utterances found.
                 if len(alternative) <= limit:
                     response = alternative
                 else:
                     response = alternative[:limit]
             else:
-                response = [alternative[0]['transcript']]
+                response = [alternative[0]["transcript"]]
         else:
             response = []
 
         return response
+
+    def _update_account_activity_ts(self):
+        self.account_repo.update_last_activity_ts(self.account.id)
+
+    def _track_account_activity(self):
```

Comment:
This is replicated in a lot of places. Can it be moved to common parent class, maybe `PublicEndpoint`? or if that's not possible make it a function.

```python
def track_account_activity(db, account):
    account_activity_repository = AccountActivityRepository(db)
    account_activity_repository.increment_activity(account)
```

Reply:
moved this and the account activity timestamp update to a function.

Reply:
They still seem to be here as members? Is there a commit missing?